### PR TITLE
Helper to get version number from package.xml

### PR DESCRIPTION
**Pull request by scpeters**
# 🎉 New feature

Follow-up to https://github.com/gazebosim/gz-math/pull/639, part of https://github.com/gazebosim/sdformat/pull/1491

## Summary

This adds a python script and a cmake helper function for getting the version number from a package.xml file.

## Test it

Test using the `version_from_package_xml` example.

1. Build and install gz-cmake from this branch
2. `cd examples/version_from_package_xml`
3. `mkdir build && cd build`
4. `cmake ..`
6. You should see the following lines in the console output, pulling the version number from the [example package.xml file](https://github.com/gazebosim/gz-cmake/blob/7c1ff61b42ae645d4dc58ccd8ca4760874702f76/examples/version_from_package_xml/package.xml#L5) 

~~~
-- PACKAGE_XML_VERSION 8.21.65
-- PACKAGE_XML_VERSION_MAJOR 8
-- PACKAGE_XML_VERSION_MINOR 21
-- PACKAGE_XML_VERSION_PATCH 65
-- gz-version_from_package_xml version 8.21.65
~~~

## Checklist
- [X] Signed all commits for DCO
- [X] Added tests
- [X] Added example and/or tutorial
- [X] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [X] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [X] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: tools/print_package_xml_version.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+if root.tag != 'package':
+    raise RuntimeError('Invalid package.xml file, root tag <%s> should be <package>' % root.tag)
+
+version_str = root.find('version').text
+
+# validate version string using regex from catkin_pkg
+# https://github.com/ros-infrastructure/catkin_pkg/blob/1.0.0/src/catkin_pkg/package_version.py#L55-L58
+match = re.match(r'^(\d+)\.(\d+)\.(\d+)$', version_str)
+if match is None:
+    raise ValueError('Invalid version string, must be int.int.int: "%s"' % version)
```
Comment (j-rivero) [inline tools/print_package_xml_version.py:35]:
```suggestion
    raise ValueError('Invalid version string, must be int.int.int: "%s"' % version_str)
```

File: tools/print_package_xml_version.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    raise RuntimeError('Expected one argument with the path to a package.xml file')
+
+file_name = sys.argv[1]
+
+doc = ET.parse(file_name)
+root = doc.getroot()
+if root.tag != 'package':
+    raise RuntimeError('Invalid package.xml file, root tag <%s> should be <package>' % root.tag)
+
+version_str = root.find('version').text
```
Comment (j-rivero) [inline tools/print_package_xml_version.py:29]:
```suggestion
if root.find('version') is None:
    raise RuntimeError('Invalid package.xml file, no <version> tag found.')

version_str = root.find('version').text
```

**Review by j-rivero — CHANGES_REQUESTED** [review]
Made some cosmetic commits, one suggestion and one real fix.

Thanks for the PR!

File: tools/print_package_xml_version.py (hunk 1 above)
Comment (scpeters) [reply tools/print_package_xml_version.py]: oops, fixed in https://github.com/gazebosim/gz-cmake/pull/456/commits/1630a52ccf234930bcfd122a2d6aa363972911a8

**Review by j-rivero — APPROVED** [review]
Good to go with green CI ! Thanks!

**Pull request merged**